frontend/src/components/common/Resource/PortForward.tsx

```diff
@@ -0,0 +1,174 @@
+          return;
+        }
+        const parsedPortForwards = JSON.parse(portForwards);
+        parsedPortForwards.push(data);
+        localStorage.setItem(
+          isPod ? PORTFORWARD_STORAGE_KEY_FOR_PODS : PORTFORWARD_STORAGE_KEY_FOR_SERVICES,
+          JSON.stringify(parsedPortForwards)
+        );
+      })
+      .catch(error => {
+        setError(error);
+        setPortForward(null);
+      });
+  }
+
+  function portForwardStopHandler() {
+    if (!portForward || !cluster) {
+      return;
+    }
+    deletePortForward(
+      cluster,
+      portForward.id,
+      isPod ? PODS_BACKEND_PORTFORWARD_KEY : SERVICE_BACKEND_PORTFORWARD_KEY
+    )
+      .then(() => {
+        setPortForward(null);
+        const id = portForward.id;
+        const portForwards = localStorage.getItem(
+          isPod ? PORTFORWARD_STORAGE_KEY_FOR_PODS : PORTFORWARD_STORAGE_KEY_FOR_SERVICES
+        );
+        if (!portForwards) {
+          return;
+        }
+        const parsedPortForwards = JSON.parse(portForwards);
+        const filteredPortForwards = parsedPortForwards.filter((item: any) => item.id !== id);
+
+        localStorage.setItem(
+          isPod ? PORTFORWARD_STORAGE_KEY_FOR_PODS : PORTFORWARD_STORAGE_KEY_FOR_SERVICES,
+          JSON.stringify(filteredPortForwards)
+        );
+      })
+      .catch(error => {
+        setError(error);
+        setPortForward(null);
+      });
+  }
+
+  React.useEffect(() => {
+    if (!cluster) {
+      return;
+    }
+    listPortForward(
+      cluster,
+      isPod ? PODS_BACKEND_PORTFORWARD_KEY : SERVICE_BACKEND_PORTFORWARD_KEY
+    ).then(result => {
+      const portforwards = result;
+      if (!portforwards || Object.entries(portforwards).length === 0) {
+        return;
+      }
+      for (const key in portforwards) {
+        const item = portforwards[key];
+        if (
+          item.namespace === namespace &&
+          (item.pod === name || item.service === name) &&
+          item.cluster === cluster
+        ) {
+          setIsCurrentPodOrServicePortForwarding(true);
+          setPortForward(portforwards[key]);
+          return;
+        }
+      }
+    });
+  }, []);
+
+  if (isPod && !isPodRunning) {
+    return null;
+  }
+
+  return !portForward ? (
+    <Box display="flex">
+      <Button onClick={handlePortForward}>{t('frequent|Port Forward')}</Button>
+      {error && (
+        <Box>
+          {
+            <Alert
+              severity="error"
+              onClose={() => {
+                setError(null);
+              }}
+            >
+              <Tooltip title={error}>
+                <Box style={{ overflow: 'hidden', textOverflow: 'ellipsis' }}>{error}</Box>
+              </Tooltip>
+            </Alert>
+          }
+        </Box>
+      )}
+    </Box>
+  ) : (
+    <>
+      {isCurrentPodOrServicePortForwarding &&
+        (portForward ? (
+          <>
+            <Button onClick={portForwardStopHandler}>Stop</Button>
```

Comment:
This button is not translated.

Reply:
These buttons should also get an aria-label as I think it's difficult for screen reader users to understand the whole context without the visual cues.

Reply:
updated